src/BPM/bond_bpm.cpp

```diff
@@ -82,7 +87,7 @@ BondBPM::~BondBPM()
   if (id_fix_dummy) modify->delete_fix(id_fix_dummy);
   if (id_fix_dummy2) modify->delete_fix(id_fix_dummy2);
   if (id_fix_update) modify->delete_fix(id_fix_update);
-  if (id_fix_bond_history) modify->delete_fix(id_fix_bond_history);
+  if (fix_bond_history) modify->delete_fix(id_fix_bond_history);
```

Comment:
Why this change?

Reply:
Previously `id_fix_bond_history` was defined in the child class right before creating the fix. I decided to consolidate the fix initialization in the parent class and moved the id assignment to the child constructor to make it's definition more apparent. 

So now there's a gap between the id definition in the constructor and the fix definition in settings. This causes an error when reading a restart file since `modify` cannot find a fix with the given id.